client/state/ui/page/reducer.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Internal dependencies
+ */
+import { SET_PAGE_STATE, RESET_PAGE_STATE } from 'state/action-types';
+
+/**
+ * Tracks the current page state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export default function( state = {}, action ) {
+	switch ( action.type ) {
+		case SET_PAGE_STATE:
+			state = Object.assign( {}, state, {
+				[ action.key ]: action.value
```

Comment:
>I think we may need at least some warning here against passing reference or a function.

I like this idea, as I agree it may be tempting for a developer to add these types of values.